comm/src/main/scala/coop/rchain/comm/protocol.scala

```diff
@@ -138,6 +138,16 @@ case class LookupMessage(proto: Protocol, timestamp: Long) extends ProtocolMessa
   */
 case class LookupResponseMessage(proto: Protocol, timestamp: Long) extends ProtocolResponse
 
+case class HandshakeMessage(proto: Protocol, timestamp: Long) extends ProtocolMessage {
+  def response(src: ProtocolNode): Option[ProtocolMessage] =
+    for {
+      h <- header
+    } yield
+        HandshakeResponseMessage(ProtocolMessage.handshakeResponse(src, h),
+          System.currentTimeMillis)
```

Comment:
Should use an explicitly-provided clock (whether injected or passed as an argument) rather than an ambient clock; otherwise testing's a big pain.

Reply:
That's true. Right now this is just standing in for a nonce (and it's not being used as time of day), but it is certainly non-reproducible.

Reply:
OK, add a TODO here to inject/pass in a nonce generator, then LGTM.

Reply:
Added a comment at the top of the file, since it's used more than just here.